webboot/webboot.go

```diff
@@ -23,9 +27,62 @@ var (
 	ipv6    = flag.Bool("ipv6", true, "use IPV6")
 )
 
+func wget(arg string) (io.ReadCloser, error) {
+	resp, err := http.Get(arg)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != 200 {
+		return nil, fmt.Errorf("non-200 HTTP status: %d", resp.StatusCode)
+	}
+	return resp.Body, nil
+}
+
+func writer(read io.ReadCloser, fileName string) (string, error) {
+	w, err := os.Create(fileName)
+	if err != nil {
+		return "", err
+	}
+
+	defer w.Close()
+
+	_, err = io.Copy(w, read)
+	return fileName, err
+}
+
+func usage() {
+	log.Printf("Only one argument: an ISO link")
+	log.Printf("Usage: %s [ARGS] URL\n", os.Args[0])
+	flag.PrintDefaults()
+	os.Exit(2)
+}
+
 func main() {
 	flag.Parse()
 
+	if flag.NArg() != 1 {
+		usage()
+	}
+
+	argURL := flag.Arg(0)
+	if argURL == "" {
+		log.Fatalln("Want a url, got empty argument")
+	}
+
 	dhclient.Request(*ifName, *timeout, *retry, *verbose, *ipv4, *ipv6)
+
+	read, err := wget(argURL)
+	if err != nil {
+		log.Fatalf("Error receiving wget: %v", err)
+	}
+
+	fileName := "fileName"
```

Comment:
why the variable? You only use it once in the next line?

Reply:
this was just for testing a general purposes as it was a work in progress and I needed it to simply test it inside of u-root to see if my webboot command works as a whole. I have made significant amount of edits afterwards